django/db/models/query.py

```diff
@@ -815,6 +815,26 @@ def complex_filter(self, filter_obj):
         else:
             return self._filter_or_exclude(None, **filter_obj)
 
+    def _combinator_query(self, combinator, *other_qs, **kwargs):
+        # Clone the query so we inherit the select list and everything
+        clone = self._clone()
+        # Clear limits and ordering, so they can be reapplied
+        clone.query.clear_ordering(True)
+        clone.query.clear_limits()
+        clone.query.combined_queries = (self.query,) + tuple(qs.query for qs in other_qs)
+        clone.query.combinator = combinator
+        clone.query.combinator_all = kwargs.pop('all', False)
+        return clone
+
+    def union(self, *other_qs, **kwargs):
```

Comment:
Would it make sense to have union_all as separate method? That way there would be no need for unbounded kwargs. If not, at least assert here that there are no other kwargs than 'all' given here.